packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -5739,3 +5710,24 @@ class RestorableRouteFuture<T> extends RestorableProperty<String?> {
 
   static NavigatorState _defaultNavigatorFinder(BuildContext context) => Navigator.of(context);
 }
+
+extension <T> on Iterable<T> {
+  T? firstWhereOrNull(bool Function(T element) test) {
```

Comment:
flutter style guide against extension, can you create a private method that takes a iterable and test function instead?